csrc/multi_tensor_adagrad.cu

```diff
@@ -10,7 +10,11 @@
 #include "multi_tensor_apply.cuh"
 #include "type_shim.h"
 
+#ifdef __HIP_PLATFORM_HCC__
+#define BLOCK_SIZE 512
+#else
 #define BLOCK_SIZE 1024
+#endif
```

Comment:
@jeffdaily Is there a reason that we want to reduce BLOCK_SIZE to 512 instead of using __launch_bounds__(1024)?  (Is there a maximum limit of the number that we can put in __launch_bounds__?)

Reply:
excellent point.  let me revert the adagrad change and try launch bounds 1024 and see what the CI result says.

Reply:
Looks like using 1024 works just as well.  Shall we merge this PR then?

Reply:
Awesome. Please go ahead.